docs/docs/concepts/auth.md

```diff
@@ -0,0 +1,291 @@
+    participant LG as LangGraph Backend
+    
+    Client->>Auth: 1. Login (username/password)
+    Auth-->>Client: 2. Return token
+    Client->>LG: 3. Request with token
+    LG->>Auth: 4. Validate token
+    Auth-->>LG: 5. Confirm validity
+    Note over LG: 6. Apply access control
+    LG-->>Client: 7. Return resources
+```
+
+Your `@auth.authenticate` handler in LangGraph handles steps 4-5, while your `@auth.on` handlers implement step 6.
+
+## Authentication
+
+Authentication in LangGraph runs as middleware on every request. Your `@auth.authenticate` handler receives request information and must:
+
+1. Validate the credentials
+2. Return user information if valid
+3. Raise an HTTP exception if invalid (or AssertionError)
+
+```python
+from langgraph_sdk import Auth
+
+auth = Auth()
+
+@auth.authenticate
+async def authenticate(headers: dict) -> Auth.types.MinimalUserDict:
+    # Validate credentials (e.g., API key, JWT token)
+    api_key = headers.get("x-api-key")
+    if not api_key or not is_valid_key(api_key):
+        raise Auth.exceptions.HTTPException(
+            status_code=401,
+            detail="Invalid API key"
+        )
+
+    # Return user info - only identity and is_authenticated are required
+    # Add any additional fields you need for authorization
+    return {
+        "identity": "user-123",        # Required: unique user identifier
+        "is_authenticated": True,      # Optional: assumed True by default
+        "permissions": ["read", "write"] # Optional: for permission-based auth
+        # You can add more custom fields if you want to implement other auth patterns
+        "role": "admin",
+        "org_id": "org-456"
+
+    }
+```
+
+The returned user information is available:
+
+- To your authorization handlers via `ctx.user`
+- In your application via `config["configuration"]["langgraph_auth_user"]`
+
+## Authorization
+
+After authentication, LangGraph calls your `@auth.on` handlers to control access to specific resources (e.g., threads, assistants, crons). These handlers can:
+
+1. Add metadata to be saved during resource creation by mutating the `value["metadata"]` dictionary directly.
+2. Filter resources by metadata during search/list or read operations by returning a filter dictionary.
+3. Raise an HTTP exception if access is denied.
+
+If you want to just implement simple user-scoped access control, you can use a single `@auth.on` handler for all resources and actions.
+
+```python
+@auth.on
+async def add_owner(ctx: Auth.types.AuthContext, value: dict):
+    """Add owner to resource metadata and filter by owner."""
+    filters = {"owner": ctx.user.identity}
+    metadata = value.setdefault("metadata", {})
+    metadata.update(filters)
+    return filters
+```
+
+### Resource-Specific Handlers
+
+You can register handlers for specific resources and actions using the `@auth.on` decorator.
+When a request is made, the most specific handler that matches that resource and action is called.
+
```

Comment:
Document the state of this configuration with words:

1. Authenticated users are able to create threads, read thread, create runs on threads
2. Only admin users are allowed to create new assistants
3. All other endpoints (e.g., e.g., delete assistant, crons, store) are disabled.